easybuild/easyconfigs/m/modin/modin-0.32.0-foss-2024a.eb

```diff
@@ -0,0 +1,44 @@
+easyblock = 'PythonBundle'
+
+name = 'modin'
+version = '0.32.0'
+
+homepage = 'https://github.com/modin-project/modin'
+description = """Modin uses Ray, Dask or Unidist to provide an effortless way to speed up your pandas notebooks, scripts, and libraries. """
+
+toolchain = {'name': 'foss', 'version': '2024a'}
+
+builddependencies = [
+    ('Cython', '3.0.10'),
+]
+
+dependencies = [
+    ('Python', '3.12.3'),
+    ('Python-bundle-PyPI', '2024.06'),
+    ('SciPy-bundle', '2024.05'),
+    ('dask', '2024.9.1'),
+    ('OpenMPI', '5.0.3'),
+    ('Ray-project', '2.37.0'),
+    ('Arrow', '17.0.0'),
+    ('mpi4py', '4.0.1')
+]
+
+download_dep_fail = True
+use_pip = True
+
+exts_list = [
+    (name, version, {
+        'checksums': ['9f68557add5e92617d006eb98bfd93d298e7d4d9932a0cd48a5e38ee4f30a134'],
+    }),
+    ('unidist', '0.7.1', {
+        'source_tmpl': '%(namelower)s-%(version)s-cp312-cp312-manylinux_2_17_%(arch)s.manylinux2014_%(arch)s.whl',
```

Comment:
Any particular reason to prefer the wheel for this one? We usually go for the source.

I was also tempted to suggest breaking this out into it's own package, but i see so far modin is the only usecase, so maybe not..? Things might change though, so i wouldn't be against putting this into it's own thing, and just have modin depend on it instead of mpi+dask+ray
https://unidist.readthedocs.io/en/latest/using_unidist/index.html#libraries-powered-by-unidist

I'm also confused as to why we need the extras here specific to mpi? it supposed dask, ray, mpi, and we have all of those as dependencies, so presumably, this should be used here? 
> unidist automatically detects which execution backends are installed and uses that for scheduling computation.
wouldn't mpi4py be enough then?

Reply:
There was some error related to copying/finding files during Cython compilation when I tried to build from source.

About the pip install extras, I'm just following the requirements in the modin installation files which specify "unidist[mpi]". I guess that only amounts to a dependency check?

Reply:
Also, Modin basically supports all three frameworks, but unidist seems to be specifically for MPI, from their setup.py:

https://github.com/modin-project/modin/blob/main/setup.py

Their readme also says that you must set both MODIN_BACKEND and UNIDIST_BACKEND to MPI if you use unidist. So it's a bit of a special thing.